### DISCUSSION TITLE
MTX_HLSVARIANT

### DISCUSSION THREAD

**Discussion by rf23** [mrlt8/docker-wyze-bridge#992] [General]
I recently restarted the bridge (likely with a new docker image) and found the HSL m3u8 stream of the camera can no longer be played on my Roku TV. Roku's video player spits out an error saying the playlist format is not recognized. 

After some digging. I found the latest version of the bridge has changed to use fmp4 instead of mpegts. Apparently Roku video player does not support fmp4 evem with its latest firmware. 

The solution is to set MTX_HLSVARIANT=mpegts. This will make the stream playlist compatible with Roku. Cheers.

**Comment by mrlt8** [comment]
Good catch! I wonder if we should default to mpegts for maximum compatibility?

**Comment by mach7** [comment]
How do I go about making this work with Home Assistant and viewing cameras on IP Camera Viewer Pro on my Rokus?

**Comment by nkm8** [comment]
Symptoms: Streaming via the m3u8s in Emby would result in freezing shortly after starting the stream. The exact duration varied, but I was never able to stream for longer than 5 minutes.

I added this to the MediaMTX config section of the Home Assistant addon. The resulting YAML looks like this:
```yaml
MEDIAMTX:
  - MTX_HLSVARIANT=mpegts
```

Result: I am now able to view cameras in Emby with unlimited duration - the config change completely resolved my issue.

I am not sure if it still makes sense to change the default, but this finally resolved my issues.

**Reply by mrlt8** [reply]
We're currently defaulting to `fmp4`, but we can switch to `mpegts` since it seems to increase compatibility:

```
# Variant of the HLS protocol to use. Available options are:
# * mpegts - uses MPEG-TS segments, for maximum compatibility.
# * fmp4 - uses fragmented MP4 segments, more efficient.
# * lowLatency - uses Low-Latency HLS.
```

https://github.com/mrlt8/docker-wyze-bridge/commit/f5dc97fd1e61a89510b20fb2b39316c45ad2ff48
